src/libutil/strutil.cpp

```diff
@@ -614,13 +614,41 @@ Strutil::split(string_view str, std::vector<string_view>& result,
 
 
 
+std::string
+Strutil::concat(string_view s, string_view t)
+{
+    size_t sl  = s.size();
+    size_t tl  = t.size();
+    size_t len = sl + tl;
+    std::unique_ptr<char[]> heap_buf;
+    char local_buf[256];
+    char* buf = local_buf;
+    if (len > sizeof(local_buf)) {
+        heap_buf.reset(new char[len]);
+        buf = heap_buf.get();
+    }
+    memcpy(buf, s.data(), sl);
+    memcpy(buf + sl, t.data(), tl);
+    return std::string(local_buf, len);
```

Comment:
You want `buf` here, not `local_buf`

Reply:
And *ahem* it looks like I was  the one who wrote the code incorrectly in the first place :(

Reply:
That will teach me an important lesson about cut and paste from untrusted sources...

Reply:
And I am similarly embarrassed about not only failing to notice it while reviewing your code, but failing TWICE MORE when putting the equivalent into OIIO, and then one final failure by actually making test cases, but that do not cover the "long string" case.